src/js/components/modals/ServiceFormModal.js

```diff
@@ -135,6 +136,13 @@ class ServiceFormModal extends mixin(StoreMixin) {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    let {state, props} = this;
+    return props.open !== nextProps.open ||
+      state.jsonMode !== nextState.jsonMode ||
+      state.errorMessages !== nextState.errorMessage;
+  }
```

Comment:
We're re-rendering if the `errorMessage` is updated (e.g. cleared) or (!) the `jsonMode` is toggled or the modal open property is toggled. Every other case should be handled by the child components.